Wrong Scanf arguments

Compiling latest version with g++ (Ubuntu 9.2.1-9ubuntu2) gives warnings:
 
```
tinyxml2/tinyxml2.cpp: In static member function ‘static bool tinyxml2::XMLUtil::ToInt(const char*, int*)’:
inyxml2/tinyxml2.cpp:671:48: warning: format ‘%x’ expects argument of type ‘unsigned int*’, but argument 3 has type ‘int*’ [-Wformat=]
  671 |     if (TIXML_SSCANF(str, IsPrefixHex(str) ? "%x" : "%d", value) == 1)
      |                                               ~^          ~~~~~
      |                                                |          |
      |                                                |          int*
      |                                                unsigned int*
      |                                               %x
tinyxml2/tinyxml2.cpp: In static member function ‘static bool tinyxml2::XMLUtil::ToInt64(const char*, int64_t*)’:
ttinyxml2/tinyxml2.cpp:741:50: warning: format ‘%llx’ expects argument of type ‘long long unsigned int*’, but argument 3 has type ‘long long int*’ [-Wformat=]
  741 |     if (TIXML_SSCANF(str, IsPrefixHex(str) ? "%llx" : "%lld", &v) == 1)
      |                                               ~~~^            ~~
      |                                                  |            |
      |                                                  |            long long int*
      |                                                  long long unsigned int*
      |                                               %llx
```

--- a/tinyxml2.cpp
+++ b/tinyxml2.cpp
@@ -610,8 +610,17 @@ void XMLUtil::ToStr( uint64_t v, char* buffer, int bufferSize )
 
 bool XMLUtil::ToInt(const char* str, int* value)
 {
-    if (TIXML_SSCANF(str, IsPrefixHex(str) ? "%x" : "%d", value) == 1) {
-        return true;
+    if (IsPrefixHex(str)) {
+        unsigned v;
+        if (TIXML_SSCANF(str, "%x", &v) == 1) {
+            *value = static_cast<int>(v);
+            return true;
+        }
+    }
+    else {
+        if (TIXML_SSCANF(str, "%d", value) == 1) {
+            return true;
+        }
     }
     return false;
 }
@@ -670,11 +679,20 @@ bool XMLUtil::ToDouble( const char* str, double* value )
 
 bool XMLUtil::ToInt64(const char* str, int64_t* value)
 {
-	long long v = 0;	// horrible syntax trick to make the compiler happy about %lld
-	if (TIXML_SSCANF(str, IsPrefixHex(str) ? "%llx" : "%lld", &v) == 1) {
-		*value = static_cast<int64_t>(v);
-		return true;
-	}
+    if (IsPrefixHex(str)) {
+        unsigned long long v = 0;	// horrible syntax trick to make the compiler happy about %llx
+        if (TIXML_SSCANF(str, "%llx", &v) == 1) {
+            *value = static_cast<int64_t>(v);
+            return true;
+        }
+    }
+    else {
+        long long v = 0;	// horrible syntax trick to make the compiler happy about %lld
+        if (TIXML_SSCANF(str, "%lld", &v) == 1) {
+            *value = static_cast<int64_t>(v);
+            return true;
+        }
+    }
 	return false;
 }
 

